### PR TITLE
Doc strings everywhere.

### PR DESCRIPTION
**Pull request by JuliaLinhart**

### CODE REVIEW

File: benchmark_utils/common.py [hunk 1]
```diff
@@ -51,13 +51,15 @@ def data_generator_sbibm(
...
         name: A task name (e.g. `'two_moons'`).
         train_size: The number of samples in the training set.
         test_size: The number of samples in the test set.
         ref_size: The number of reference posteriors.
         n_per_ref: The number of samples per reference posteriors.
 
     Returns:
+    --------
         A dictionary with training, test and reference sets as well as the parameters
         prior.
```
Comment (agramfort) [inline benchmark_utils/common.py]: can you stick to numpydoc style?

```suggestion
    Parameters
    ----------
    name: str
        A task name (e.g. `'two_moons'`).
    train_size: int
        The number of samples in the training set.
    test_size: int
        The number of samples in the test set.
    ref_size: int
        The number of reference posteriors.
    n_per_ref: int
        The number of samples per reference posteriors.

    Returns
    -------
    dict
        A dictionary with training, test and reference sets as well as the parameters
        prior.
```

**Review by agramfort — COMMENTED** [review]
see https://numpydoc.readthedocs.io/en/latest/format.html#parameters

you can run pydocstyle http://www.pydocstyle.org/en/stable/
or your code to help

**Comment by francois-rozet** [comment]
In terms of docstrings, I prefer the [Google style](https://google.github.io/styleguide/pyguide.html) (what was used previously) to the NumPy style. I find the former much more readable and less verbose (less vertical space). I believe the Google style is more popular than the NumPy style. Note that torch, sbi and lampe use the Google style already.

Also, one of the purposes of type annotations is to not have to write them in the docstring. Having them both in the signature and the docstring is redundant and a source of inconsistency.

**Comment by agramfort** [comment]
to me it's convention to adopt in the context. All benchopt code/benchmarks
follow numpy standard.

but I won't fight for this to death

**Comment by francois-rozet** [comment]
If it's the convention of Benchopt then let's keep it.

**Comment by JuliaLinhart** [comment]
If you're okay with it, I'll merge and close this PR.

**Pull request merged**